web_src/js/components/RepoContributors.vue

```diff
@@ -0,0 +1,249 @@
+      );
+      const [cooefficient, exp] = maxValue
+        .toExponential()
+        .split('e')
+        .map(Number);
+      if (cooefficient % 1 === 0) {
+        return maxValue;
+      }
+      return (1 - (cooefficient % 1)) * 10 ** exp + maxValue;
+    },
+    additions(data) {
+      return Object.values(data).reduce((acc, item) => {
+        return acc + item.additions;
+      }, 0);
+    },
+    deletions(data) {
+      return Object.values(data).reduce((acc, item) => {
+        return acc + item.deletions;
+      }, 0);
+    },
+
+    toGraphData(data, type) {
+      return {
+        datasets: [
+          {
+            data: data.map((i) => {
+              return {x: i.week, y: i[this.type]};
+            }),
+            pointRadius: 0,
+            pointHitRadius: 0,
+            fill: 'start',
+            backgroundColor:
+              type === 'main' ?
+                'rgba(137, 191, 154, 0.6)' :
+                'rgba(96, 153, 38, 0.7)',
+            borderWidth: 0,
+            tension: 0.3,
+          },
+        ],
+      };
+    },
+
+    updateOtherCharts(event) {
+      const minVal = event.chart.options.scales.x.min;
+      const maxVal = event.chart.options.scales.x.max;
+
+      for (const instance of Object.values(Chart.instances)) {
+        if (instance !== event.chart) {
+          instance.options.scales.x.min = minVal;
+          instance.options.scales.x.max = maxVal;
+          instance.update();
+        }
+      }
+    },
+
+    getOptions(type) {
+      return {
+        responsive: true,
+        animation: false,
+        onClick: (e) => {
+          if (type === 'main') {
+            e.chart.resetZoom();
+          }
+        },
+        plugins: {
+          legend: {
+            display: false,
+          },
+          zoom: {
+            pan: {
+              enabled: true,
+              mode: 'x',
+              threshold: 20,
+
+              onPan: this.updateOtherCharts,
+            },
+            limits: {
+              x: {
+                min: 'original',
+                max: 'original',
+                minRange: 1000000000,
+              },
+            },
+            zoom: {
+              wheel: {
+                enabled: type === 'main',
+              },
+              pinch: {
+                enabled: type === 'main',
+              },
+              mode: 'x',
+
+              onZoomComplete: this.updateOtherCharts,
+            },
+          },
+        },
+        scales: {
+          x: {
+            type: 'time',
+            grid: {
+              display: false,
+            },
+            time: {
+              minUnit: 'day',
+            },
+          },
+          y: {
+            min: 0,
+            max: this.maxMainGraph(),
+          },
+        },
+      };
+    },
+  },
+};
+
+export function initRepoContributorsChart() {
+  const el = document.getElementById('repo-contributors-chart');
+  if (el) {
+    createApp(sfc).mount(el);
+  }
+}
```

Comment:
This kind of init will not code-split the component from the `index.js` bundle, which we really need to do. I think the easiest method would to move this init to another file and then `await import` the `sfc` exported from this component.

FWIW, vue also has [async components](https://vuejs.org/guide/components/async.html), but I guess we don't really need  those in this case.

Reply:
I have tried code-split as you said but it didn't work. Can you look at [this diff](https://github.com/sahinakkaya/gitea/commit/28ecebb2f3bc745f66ff908778a8c030980bd35c) and tell me what's wrong?)

Reply:
The `await import` import needs to be in the called function and "guarded" by the `if` before the prevent loading when not needed. I see you used some `<script>`, I don't think that is necessary.

Reply:
Tried this one too. It still doesn't work :( Maybe someone more knowledgeable can handle this? 
Here is what I have tried: https://github.com/sahinakkaya/gitea/commit/b434a7af710bfa4e20288e4571bb23bc7765daf5

Reply:
And the error was this on the console: `component is missing template or render function`